### PR TITLE
glance: Restrict image_location metadata (bsc#1023507)

### PR DESCRIPTION
**Pull request by aplanas**
When `show_multiple_locations` is enabled in Glance, any user
can rewrite the metadata information for locations, causing
a securiy breach.

This patch limit to admin the editing of image_location metadata
when this feature is enabled.

### CODE REVIEW

**Comment by aplanas** [comment]
[P2: Bug 1023507 - VUL-0: openstack-glance: Users of Glance may be able to replace active image data (OSSN 0065)](https://trello.com/c/XIKMt2au/345-p2-bug-1023507-vul-0-openstack-glance-users-of-glance-may-be-able-to-replace-active-image-data-ossn-0065)

**Comment by cmurphy** [comment]
If anyone has a custom policy.json for glance this will completely break their policies, which is a major security concern.

Could we somehow do a find and replace in the existing file instead of managing the whole file?

Could this maybe be fixed in the package instead? (not sure how upgrading would work if they've modified the file)

Could we just create some messaging to users that they should do this themselves?

File: chef/cookbooks/glance/recipes/api.rb [hunk 1]
```diff
@@ -103,6 +103,26 @@
...
 
+# If `show_multiple_locations` is enabled we need to update
+# policy.json in a more restrictive way.  Only admin (or image owner)
+# can now update the `image_location` metadate field.
+#
+# For more information check bsc#1023507, and upstream bug:
+# https://bugs.launchpad.net/ossn/+bug/1549483
+#
+# NOTE(aplanas) -- policy.json needs to be synchronized to current
+# version of OpenStack.
```
Comment (cmurphy) [inline chef/cookbooks/glance/recipes/api.rb]: This is super important and super painful. This comment should maybe include the date or release that it was last modified to make it easier to track whether it's current.

**Comment by aplanas** [comment]
The other option is to move it into the package, that have the problem that will use, by default, the most restrictive version of the policy

**Comment by aplanas** [comment]
@cmurphy lets drop this and try to put it in the package for the constrained version!

Thanks!

**Pull request closed without merging**